include/arch/arm/cortex_r/asm_inline_gcc.h

```diff
@@ -0,0 +1,158 @@
+/* ARM Cortex-M GCC specific public inline assembler functions and macros */
```

Comment:
IMHO with some modifications in _arch_irq_lock()/_arch_irq_unlock() this file could be common for Cortex-M and R.